web/src/user_group_edit.js

```diff
@@ -114,6 +114,24 @@ export function handle_member_edit_event(group_id) {
     const members = [...group.members];
     user_group_edit_members.update_member_list_widget(group_id, members);
 
+    // update display of group-row.
+    user_group_settings_ui.redraw_user_group_list();
```

Comment:
`ListWidget` also maintains the list of items and that needs to be updated. If we not update the group list maintained by the `ListWidget` it would remove the newly subscribed group or would again show some group for which the user was unsubscribed during some re-render due to interaction with search bar.
So Listwidget when we updates data maintained by `ListWidget` it automatically updates the list of items shown on DOM based on updated data.

I've made changes to optimize this and make calls to `user_group_settings_ui.redraw_user_group_list()` only when needed and also added a comment to explain the changes.